pkgs/games/anki/default.nix

```diff
@@ -1,199 +1,372 @@
+        exit 0
+      ;;
+      *)
+        echo "Unrecognized git: $@"
+        exit 1
+      ;;
+    esac
+  '';
+
+  # we built a bunch of rust packages above, so we can just not build them when
+  # the runner tries to.
+  # We mock this out with very specific matching so that if, for example, the
+  # build system adds a new '--features' flag, we'll error out here on update
+  # and know to then update our rust builds above.
+  fakeCargo = writeShellScriptBin "cargo" ''
+    #!${bash}/bin/bash
+
+    case "$*" in
+      "run -p configure")
+        exec ${anki-configurator}/bin/configure
+        ;;
+      "build --locked -p configure -p runner" | "build --locked -p rsbridge" | "build --release --locked -p anki_i18n")
+        exit 0
+        ;;
+      "build --release --locked -p rsbridge --features rustls" | "build --locked -p archives --features rustls")
+        exit 0
+        ;;
+      *build*)
+        1>&2 echo "Not building: cargo $@"
+        exit 1
+        ;;
+      *)
+        echo "Unknown cargo command: $@"
+        exit 1
+        ;;
+    esac
+  '';
+
+  # We don't want to run pip-sync, it does network-io
+  fakePipSync = writeShellScriptBin "pip-sync" ''
+    exit 0
+  '';
+
+  # archives is the part of their build system that downloads and checks shasums. We don't want their system downloading stuff, so we mock it out entirely
+  fakeArchive = writeShellScriptBin "archives" ''
+    1>&2 echo "Not archives: $@"
+    exit 0
+  '';
+
+  # corepack is some nodejs thing that gets called, idk, seems bad
+  fakeCorepack = writeShellScriptBin "corepack" ''
+    1>&2 echo "Fake corepack: $@"
+    exit 0
+  '';
+
+  offlineYarn = writeShellScriptBin "yarn" ''
+    [[ "$1" == "install" ]] && exit 0
+    exec ${yarn}/bin/yarn --offline "$@"
+  '';
+
+  pyEnv = buildEnv {
+    name = "anki-pyenv-${version}";
+    paths = with python39.pkgs; [
+      pip
+      fakePipSync
+      anki-build-python
     ];
-    sha256 = sha256-pkg;
+    pathsToLink = [ "/bin" ];
   };
 
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    sha256 = "sha256-bAtmMGWi5ETIidFFnG3jzJg2mSBnH5ONO2/Lr9A3PpQ=";
+  };
+
+  # https://discourse.nixos.org/t/mkyarnpackage-lockfile-has-incorrect-entry/21586/3
+  anki-nodemodules = stdenv.mkDerivation {
+    pname = "anki-nodemodules";
+    inherit version src yarnOfflineCache;
+
+    nativeBuildInputs = [
+      fixup_yarn_lock
+      yarn
+      nodejs-slim
+    ];
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+      yarn run postinstall --offline
+    '';
+
+    installPhase = ''
+      mv node_modules $out
+    '';
+  };
+
+  # Mask ninja's setup hooks, we just want the binary, not the implicit 'ninja install'
+  ninja-bin = buildEnv {
```

Comment:
You can use `dontUseNinjaInstall = false;` instead.
See https://nixos.org/manual/nixpkgs/unstable/#ninja

Reply:
done, thanks!